codec-sockjs/src/test/java/io/netty/handler/codec/sockjs/handler/PollingSessionStateTest.java

```diff
@@ -0,0 +1,107 @@
+/*
+ * Copyright 2014 The Netty Project
+ *
+ * The Netty Project licenses this file to you under the Apache License, version
+ * 2.0 (the "License"); you may not use this file except in compliance with the
+ * License. You may obtain a copy of the License at:
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+package io.netty.handler.codec.sockjs.handler;
+
+
+import io.netty.channel.Channel;
+import io.netty.channel.ChannelHandlerContext;
+import io.netty.channel.DefaultEventLoop;
+import io.netty.handler.codec.sockjs.SockJsConfig;
+import io.netty.handler.codec.sockjs.SockJsService;
+import io.netty.handler.codec.sockjs.SockJsSessionContext;
+import io.netty.handler.codec.sockjs.protocol.HeartbeatFrame;
+import org.junit.Test;
+import java.util.concurrent.ConcurrentHashMap;
+import java.util.concurrent.Executors;
+
+import static org.mockito.Matchers.any;
+import static org.mockito.Mockito.atLeastOnce;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.when;
+
+public class PollingSessionStateTest {
+
+    @Test
+    public void onOpen() {
+        final SockJsService service = mock(SockJsService.class);
+        final SockJsConfig config = SockJsConfig.withPrefix("/echo").sessionTimeout(1).build();
+        final SockJsSessionContext sessionContext = mock(SockJsSessionContext.class);
+        final ChannelHandlerContext ctx = inactiveCtx();
+
+        pollingSessionState(service, config, ctx).onConnect(ctx, sessionContext);
+        verify(service).onOpen(sessionContext);
+    }
+
+    @Test
+    public void onClose() throws InterruptedException {
+        final SockJsService service = mock(SockJsService.class);
+        final SockJsConfig config = SockJsConfig.withPrefix("/echo").sessionTimeout(1).build();
+        final ChannelHandlerContext ctx = inactiveCtx();
+
+        pollingSessionState(service, config, ctx).onConnect(ctx, mock(SockJsSessionContext.class));
+        Thread.sleep(10);
```

Comment:
![CRITICAL](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-critical.png) Remove this use of "Thread.sleep()". [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://garage.netty.io/sonarqube/coding_rules#rule_key=squid%3AS2925)